Tenure fixes

jessicacamacho opened the pull request:
Hi @smmaurer,

Regarding https://github.com/urbansim/consulting-planning/issues/339, this PR adds the changes to fix tenure imbalances and enables using the `tenure` variable for `units` and `households`. 

To show evolution of tenure, I created the `Implementing Housing Tenure` notebook, which runs the model for scenarios 4 and 5 and retrieves the tenure values for households and units using the new `units_tenure_track` and `households_tenure_track` injectables. The notebook plots evolution of tenure split for these scenarios, and shows that all renter households were assigned to renter units only, all owner households were assigned to owned units, and no households have `null` or `-1` values for `unit_id`.

Please let me know if you have any questions, or would like me to make any changes.

Thanks!

Jessica

jessicacamacho committed:
Turns alternative filters on, so that owner and renter hlcm models only choose from owned and rented units

(cherry picked from commit 3d9cfa13f5eef63d419368f03812b46e4066a061)

jessicacamacho committed:
Adds step to save tenure values for households and units every simulation year

(cherry picked from commit 4b6e464ec9dfb7c82fa274439ccb0ee51206d4bd)

jessicacamacho committed:
Adds functions to fix the issue with alt_filters in the hlcm simulations.

(cherry picked from commit ceb7c8fdf97903d9d9b8c38f7b80717927b241ed)

jessicacamacho committed:
Changes step order, placing assign_tenure_to_new_units after transition

(cherry picked from commit ca20996699b6ff97cb2bfea95e55c6da778c255f)

jessicacamacho committed:
Fixes bug in function to assign tenure considering min required units

(cherry picked from commit 40d6293d3eefb4bd909f8c1b56b82ffe78d5e290)

jessicacamacho committed:
Modifies add_population, removing check when adding population to taz_df

(cherry picked from commit dfe813acda40b01b9ce103dca1a0644774bdbf6f)

jessicacamacho committed:
Numpy version bug fixes

jessicacamacho committed:
Adds Notebook to analyze evolution of tenure

jessicacamacho committed:
Changes to meet pep8 requirements

smmaurer commented:
Hi @jessicacamacho, thank you -- this looks great. I'm going to run a full simulation using this branch, to make sure that it works on my machine and that i understand what it's doing. But from looking through the code, everything looks good!

jessicacamacho committed:
Updates to tenure notebook to run simulations in sequence

jessicacamacho committed:
Updates to tenure notebook to run simulations in sequence

jessicacamacho committed:
adds tenure folder in runs to export tenure data for notebook

jessicacamacho commented:
Hi @smmaurer!

I have added a few commits to update the Jupyter notebook and added the `runs/tenure` folder to save and read tenure variables from simulations that get initiated by the notebook. This new notebook shows the results of scenarios 4 and 5 up to 2050.

 Let me know if you have any questions.

Thanks!!

Jessica
